racket/collects/net/http-client.rkt

```diff
@@ -237,6 +248,60 @@
   (http-conn-open! hc host-bs #:ssl? ssl? #:port port)
   hc)
 
+(define (http-conn-CONNECT-tunnel proxy-host proxy-port target-host target-port #:ssl? [ssl? #f])
+  (define hc (http-conn-open proxy-host #:port proxy-port #:ssl? #f))
+  (define connect-string (format "~a:~a" target-host target-port))
+  ; (log-net/url-info "http-conn-CONNECT-tunnel tunnel to ~s for ~s" connect-string (url->string url))
+  (http-conn-send! hc #:method "CONNECT" connect-string #:headers
+                   (list (format "Host: ~a" connect-string)
+                         "Proxy-Connection: Keep-Alive"
+                         "Connection: Keep-Alive"))
+
+  (let ((tunnel-status (http-conn-status! hc))
+        (tunnel-headers (http-conn-headers! hc)))
+    (unless (regexp-match "^HTTP[^ ]* +2" tunnel-status)
+      (error 'make-ports "HTTP CONNECT failed: ~a" tunnel-status)))
+
+  ;; SSL secure the ports
+  (match-define (http-conn _ _ _ t:to t:from _) hc)
+  (cond [(not ssl?) ; it's just a tunnel... no ssl
+         (define abandon-p (lambda (p) ((http-conn-abandon-p hc) p)))
+         (values ssl? t:from t:to abandon-p)]
+
+        [(osx-old-openssl?)
+         (http-conn-close! hc)
+         ;; osx-old-openssl? doesn't support ports->...ssl-ports
+         (error 'http-conn-open! "osx-old-openssl? does not support a ports->...ssl-ports function")]
```

Comment:
The `http-conn` hasn't thrown an error. I can't move to the next part of the function, since I'm not talking to the target host. I need to abort. Either throw this `(error ...)` or keep the whole thing in an escape continuation. Also, if the caller were to chose not to use `http-conn-CONNECT-tunnel`, and used `tcp-connect` instead; they would get an error thrown on failure.

Would you suggest a better way to handle this error?